### PR TITLE
Fixes for ProcessTest on Windows

### PR DESCRIPTION
**Pull request by henricm**
* Checking more error codes on whitespace tests.
* Using findstr instead of type to enable stdin/out pipes.

### CODE REVIEW

File: mcs/class/System/Test/System.Diagnostics/ProcessTest.cs [hunk 1]
```diff
@@ -284,7 +284,9 @@ public void Start1_FileName_Whitespace ()
 				Assert.AreEqual (-2147467259, ex.ErrorCode, "#3");
 				Assert.IsNull (ex.InnerException, "#4");
 				Assert.IsNotNull (ex.Message, "#5");
-				Assert.AreEqual (2, ex.NativeErrorCode, "#6");
+				// On windows we get ACCESS_DENIED (5) instead of FILE_NOT_FOUND (2) and .NET
+				// gives ERROR_INVALID_PARAMETER (87).
+				Assert.IsTrue (ex.NativeErrorCode == 2 || ex.NativeErrorCode == 5 || ex.NativeErrorCode == 87, "#6");
```
Comment (akoeplinger) [inline mcs/class/System/Test/System.Diagnostics/ProcessTest.cs]: do we know if these behavior differences between .net / mono on windows are expected?

**Comment by henricm** [comment]
I haven't investigated too deeply. We considered this an edge case where the most important is that an error is generated at all. Proper fix would probably include making sure Mono returns ERROR_INVALID_PARAMETER is returned on all platforms so it matches how .NET behaves.

**Comment by akoeplinger** [comment]
Ok. Can you please file a bugzilla so this is at least tracked somewhere and add a link to the bug as a comment?

Feel free to merge then.

**Comment by henricm** [comment]
Sounds like a good plan! Created an issue about aligning the NativeErrorCode for the Win32Exception generated in this case:
https://bugzilla.xamarin.com/show_bug.cgi?id=44514

**Pull request merged**